services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -690,11 +726,15 @@ private PermissionTicketToken verifyPermissionTicket(KeycloakAuthorizationReques
 
         PermissionTicketToken ticket = request.getKeycloakSession().tokens().decode(ticketString, PermissionTicketToken.class);
         if (ticket == null) {
-            throw new CorsErrorResponseException(request.getCors(), "invalid_ticket", "Ticket verification failed", Status.FORBIDDEN);
+            CorsErrorResponseException ticketVerificationException = new CorsErrorResponseException(request.getCors(), "invalid_ticket", "Ticket verification failed", Status.FORBIDDEN);
+            fireErrorEvent(request.getEvent(), Errors.TICKET_VERIFICATION_FAILED, ticketVerificationException);
```

Comment:
I would just use `Errors.INVALID_PERMISSION_TICKET`. The details should be enough to understand what happened.

Reply:
agreed